pkg/apis/kubermatic/v1/cluster.go

```diff
@@ -296,15 +296,25 @@ func (c ClusterSpec) IsKubeLBEnabled() bool {
 func (c ClusterSpec) GetVersionConditions() []ConditionType {
 	conditions := []ConditionType{}
 
-	if c.Features[ClusterFeatureExternalCloudProvider] {
-		conditions = append(conditions, ExternalCloudProviderCondition)
-	} else {
-		conditions = append(conditions, InTreeCloudProviderCondition)
+	// we will not return the external/internal provider condition for providers that do not
+	// have a CCM.
+	if c.HasCCM() {
+		if c.Features[ClusterFeatureExternalCloudProvider] {
+			conditions = append(conditions, ExternalCloudProviderCondition)
+		} else {
+			conditions = append(conditions, InTreeCloudProviderCondition)
+		}
 	}
 
 	return conditions
 }
 
+// HasCCM returns whether there _is_ a CCM (internal or external) to be deployed.
+func (c ClusterSpec) HasCCM() bool {
+	return c.Cloud.AWS != nil || c.Cloud.Azure != nil || c.Cloud.Openstack != nil || c.Cloud.Hetzner != nil || c.Cloud.GCP != nil || c.Cloud.Anexia != nil || c.Cloud.VSphere != nil || c.Cloud.Kubevirt != nil || c.Cloud.Digitalocean != nil
+
+}
```

Comment:
Can this somehow be defined using one of the other existing functions? Like

HasCCM() {
   return DeploymentReconciler() != nil
}

or something.

Reply:
I looked into it but I'm not happy generating the whole reconciler every time. I was surprised we dont have this kind of function, but I think this should be the way to go tbh.